src/main/java/com/google/devtools/build/lib/remote/ByteStreamUploader.java

```diff
@@ -87,6 +89,7 @@
 
   @GuardedBy("lock")
   private final Map<HashCode, ListenableFuture<Void>> uploadsInProgress = new HashMap<>();
+  private final ClientInterceptor[] interceptors;
```

Comment:
I'd prefer to use ImmutableList here instead of array. One problem with using arrays is that anyone with a reference to the array can modify the contents asynchronously. Technically, ensuring the safety of this code requires reviewing all the callers to check if they're doing anything sinister. Even if it's unlikely that anyone would do so intentionally, it wouldn't be obvious if anyone is doing it accidentally.

Reply:
Changed this one to an `ImmutableList` as suggested.
There was also a similar bit in `RemoteServerCapability`, for which I opted for making a copy of the array instead, as it makes it easier to pass it directly to the `withInterceptors` method, that only accepts an array.